src/manage.c

```diff
@@ -3234,14 +3234,20 @@ fork_cve_scan_handler (task_t task, target_t target)
       exit (1);
     }
 
+  exclude_hosts = target_exclude_hosts (target);
+
   reset_task (task);
   set_task_start_time_epoch (task, time (NULL));
   set_scan_start_time_epoch (global_current_report, time (NULL));
 
   /* Add the results. */
 
   gvm_hosts = gvm_hosts_new (hosts);
+  gvm_hosts_exclude (gvm_hosts, exclude_hosts ?: "");
+
   free (hosts);
+  free (exclude_hosts);
```

Comment:
Do you think it makes sense to add error checking here ?

```suggestion
  free (hosts);
  
  if (gvm_hosts_exclude (gvm_hosts, exclude_hosts ?: "") < 0)
    {
      set_task_interrupted (task,
                              "Failed to exclude hosts."
                              "  Interrupting scan.");      
      set_report_scan_run_status (global_current_report, TASK_STATUS_INTERRUPTED);
      gvm_hosts_free (gvm_hosts);
      free (exclude_hosts);
      gvm_close_sentry ();
      exit(1);
    }
  free (exclude_hosts);
```